var/spack/repos/builtin/packages/curl/package.py

```diff
@@ -386,6 +386,10 @@ def configure_args(self):
         args += self.with_or_without("libssh", activation_value="prefix")
         args += self.enable_or_disable("ldap")
 
+        # https://github.com/spack/spack/issues/27709
+        if self.spec.compiler.name in ["intel", "oneapi"]:
+            args.append("CFLAGS=-we147")
```

Comment:
According to what I read, this can override the CFLAGS environment variable if it is set, which is why `flag_handler` is attractive: `curl` is an `AutotoolsPackage`, so you can explicitly export that flag as a `configure` argument like:

```
def flag_handler(self, name, flags):
    build_system_flags = []
    if name == "cflags" and self.spec.compiler.name in ["intel", "oneapi"]:
        build_system_flags = ["-we147"]
    # First arg is injected flags - the default - and 3rd is for exporting 
    return flags, None, build_system_flags
```

this should have the same effect as what you did, but avoid potential conflicts in the future (e.g. makes it more explicit if people want to mix in other flags what the behavior is).

Is that what your `flag_handler` (which didn't work) looked like?

I should note that currently `curl` does not define a `flag_handler`, so this isn't strictly a bug (any compiler flags specified in `compilers.yaml` or as part of the spec would be injected into the wrapper and not expressed as an env variable), so this concern is more about organization.

Reply:
Hmm unfortunately that didn't work:
```
$ git diff
diff --git a/var/spack/repos/builtin/packages/curl/package.py b/var/spack/repos/builtin/packages/curl/package.py
index f8a50829ac..0753b7cdcd 100644
--- a/var/spack/repos/builtin/packages/curl/package.py
+++ b/var/spack/repos/builtin/packages/curl/package.py
@@ -332,6 +332,12 @@ def command(self):


 class AutotoolsBuilder(AutotoolsBuilder):
+    def flag_handler(self, name, flags):
+        build_system_flags = []
+        if name == "cflags" and self.spec.compiler.name in ["intel", "oneapi"]:
+            build_system_flags = ["-we147"]
+        return flags, None, build_system_flags
+
     def configure_args(self):
         spec = self.spec
```
I get the error I was hoping to fix:
```
==> Error: ProcessError: Command exited with status 1:
    '/p/work1/heinzell/spack-stack-fix_python_ssl-test2/cache/build_stage/spack-stage-curl-8.1.2-b244hm5pwdjvx2ni7rbjoirjysy4b47p/spack-src/configure' '--prefix=/p/work1/heinzell/spack-stack-fix_python_ssl-test2/envs/curltest/install/intel/2021.5.0/curl-8.1.2-b244hm5' '--with-zlib=/p/work1/heinzell/spack-stack-fix_python_ssl-test2/envs/curltest/install/intel/2021.5.0/zlib-1.2.13-r4fjjjh' '--without-brotli' '--without-libgsasl' '--without-libpsl' '--without-zstd' '--enable-shared' '--enable-static' '--without-ca-bundle' '--without-ca-path' '--with-ca-fallback' '--without-gssapi' '--without-gnutls' '--without-mbedtls' '--without-nss' '--with-openssl=/p/work1/heinzell/spack-stack-fix_python_ssl-test2/envs/curltest/install/intel/2021.5.0/openssl-1.1.1u-qup5efn' '--without-secure-transport' '--without-sspi' '--without-libidn2' '--without-librtmp' '--without-nghttp2' '--without-libssh2' '--without-libssh' '--disable-ldap'

1 error found in build log:
     145    configure: compiler options added: -O2
     146    checking if compiler accepts strict warning options... yes
     147    configure: compiler options added: -Wno-system-headers
     148    checking if compiler halts on compilation errors... yes
     149    checking if compiler halts on negative sized arrays... yes
     150    checking if compiler halts on function prototype mismatch... no
  >> 151    configure: error: compiler does not halt on function prototype mismatch.

See build log for details:
  /p/work1/heinzell/spack-stack-fix_python_ssl-test2/cache/build_stage/spack-stage-curl-8.1.2-b244hm5pwdjvx2ni7rbjoirjysy4b47p/spack-build-out.txt
```
Our fork is currently on
```
* **Spack:** 0.21.0.dev0 (1c9041bcf17f7ff5ae36e29a0b5c4a1c7297f421)
* **Python:** 3.6.8
* **Platform:** linux-rhel8-zen3
* **Concretizer:** clingo
```
could this be an issue?

Reply:
You have to put the `flag_handler` under the `Package` class definition, not the `Builder` class definition.

Reply:
That works! I pushed the update. Thanks for your help.

Reply:
Thanks for fixing the issue!